src/server/systems/ServerGameController.cpp

```diff
@@ -83,9 +90,19 @@ void ServerGameController::postUpdate(sf::Int32 update_time) {
 void ServerGameController::setNetworkState() {
     GameState game_state = {
         .object = m_gameObjectController->getGameStateObject(),
+        .event = m_gameStateEvent,
         .core = m_gameStateCore,
     };
     m_networkingServer->setState(game_state);
+    m_gameStateEvent.collisions.clear();
+}
+
+void ServerGameController::handleCollisionEvent(std::shared_ptr<Event> event) {
+    std::shared_ptr<CollisionEvent> collision_event =
+        std::dynamic_pointer_cast<CollisionEvent>(event);
+
+    // save collision data
```

Comment:
is there no need to take a lock here?

Reply:
No, this is just date passed in to this function